### ISSUE TITLE
torch.sum bug

### ISSUE THREAD

**Issue by LU4E** [facebookresearch/maskrcnn-benchmark#109]
## ❓ Questions and Help
There is a bug in my processing of adding a loss to the mask-rcnn
first, there is a issue in the processing of turn int to float, sometime 1.0 will turn to some float value like 0.2345, which less than one. it is wired.
so i print the value of variable to analysis, and the problem always in the processing of turning cpu.int to cuda.float. so I turn cpu.int to cpu.float, then to cuda.float. it works. but some time the stile have some problem:

```
gt sum is tensor([1.3014e+05, 9.0000e+00, 6.0100e+02, 3.2500e+02], device='cuda:0')
sg is tensor([6.3204e-35, 5.1167e-35, 7.5241e-35, 6.3204e-35], device='cuda:0')

```

secend, another prolem is like this:

![2018-11-05 10-49-06](https://user-images.githubusercontent.com/12681953/47975589-69184480-e0e9-11e8-8699-23a0bf0a6504.png)

when i print the value of seg4loss, the sum_gt will be affected:

![2018-11-05 10-49-58](https://user-images.githubusercontent.com/12681953/47975787-4b97aa80-e0ea-11e8-98ca-118d4024ee9c.png)

the torch sum will be the last seg4loss i print, i check it three times, each with seg4loss[0], seg4loss[1],seg4loss[2], the sum_gt will change, which is the same with the print value.

when i delete the print code:

![2018-11-05 10-50-22](https://user-images.githubusercontent.com/12681953/47975838-9e716200-e0ea-11e8-850c-80b2a607796e.png)

it works well, i don't know why:

![2018-11-05 10-51-28](https://user-images.githubusercontent.com/12681953/47975854-afba6e80-e0ea-11e8-9888-c40156c4284d.png)

third: the softmax output always nan,  and i use the torch.clamp to avoid it. But i don't think that is a good idea, i will check the code again to find out the source of problem. by the way have anyone encountered this situation?

**Comment by LU4E** [comment]
OK, This bug has been confirmed.
all of the code works well on the pytorch 0.4 , python 3.6.
but it have some bug i can't fix on the pytorh1.0.

**Comment by fmassa** [comment]
What's your PyTorch version? Can you print the output of
```python
import torch
print(torch.__version__)
```
and paste here?

I think you might be hitting this bug https://github.com/pytorch/pytorch/issues/13209, which was fixed a few days ago in this PR https://github.com/pytorch/pytorch/pull/13211

**Comment by LU4E** [comment]
```
/home/jason/anaconda3/envs/pytorch/bin/python /home/jason/github/maskrcnn-benchmark/datacheck.py
1.0.0.dev20181028

Process finished with exit code 0
```

this is the version which has bug.
I debug this block of code which i want to add on the pytorch0.4, and it works well.

**Comment by fmassa** [comment]
Yes, this version has the `torch.sum` bug.
Update your pytorch installation and the bug should disappear, as it was recently fixed and the fix is available in the latest nightly.

I'm closing the issue as I think this bug has already been addressed, but let me know if it doesn't solve the issue for you.

**Comment by LU4E** [comment]
I can't use the conda update order to update the pytorch-nightly, do i have to remove the pytorch-nightly, and install it again?

**Comment by akshaygadipatil** [comment]
@fmassa, I am using Pytorch 1.0.0.dev20181116 version and I am using my own dataset to train. There are no errors, so I assume my data format is consistent with coco format that is fed to the network.
However, I get  'nan' values in my loss, please see below

![image](https://user-images.githubusercontent.com/22264995/48737671-85e16b80-ec03-11e8-9dbd-afbb0e9075d9.png)

Spent quite some time to figure out but can't understand why this is happening. What could be the reason?
Thanks!

**Comment by fmassa** [comment]
Are you following the learning rate rules for different number of GPUs?

**Comment by akshaygadipatil** [comment]
oops, no. Can you point out where that part is? I am using 2 GPU's btw.

**Comment by fmassa** [comment]
In the README, check the single GPU training part

**Comment by akshaygadipatil** [comment]
I am following the instructions all along:
![image](https://user-images.githubusercontent.com/22264995/48871895-77c65300-ed9b-11e8-806d-42e402d326e6.png) 
There is SOLVER.STEPS as well in the command but I didn't show in the above image.

Still getting 'nan'. 
![image](https://user-images.githubusercontent.com/22264995/48872002-1357c380-ed9c-11e8-8b6f-ea291cdca60d.png)
The data format doesn't seem to be an issue, else it would have thrown an error before starting training. What could be the cause?
Thanks.

**Comment by fmassa** [comment]
Can you try running it on a standard COCO dataset for training to see if you also have nan?
If no, then it's probably an issue with your data.

**Comment by akshaygadipatil** [comment]
I found the problem.
It was an issue with my data format. Now it runs.

@fmassa, thanks very much for your constant help!
